Add own rank to leaderboard overview

GODrums opened the pull request:
### Motivation
<!-- Explain why this change is necessary. What problem does it solve? -->
<!-- Link to the issue this PR addresses (e.g., `Fixes #123`, `Closes #123`, etc.) -->
We want to make it as easy as possible for users to find themselves in the leaderboard.

### Description
<!-- Provide a brief summary of the changes. -->
This PR adds a new entry on top of the leaderboard representing the loggedIn-user. If the user does not appear in the leaderboard yet, it provides a default replacement entry with a rank of #entries + 1.

### Testing Instructions
<!-- Explain how to test the changes made in this PR. -->
- run `application-server` and `webapp`
- login with Github and go to the homepage
- check out different leaderboards to see if your entry is on top of the leaderboard and displays correct data

### Screenshots (if applicable)
<!-- Attach screenshots here. -->
![image](https://github.com/user-attachments/assets/aad668c7-614e-4f94-9c1e-164432f2ad71)

### Checklist

#### General

- [x] PR title is clear and descriptive
- [x] PR description explains the purpose and changes
- [x] Code follows project coding standards
- [x] Self-review of the code has been done
- [x] Changes have been tested locally
- [x] Screenshots have been attached (if applicable)
- [ ] Documentation has been updated (if applicable)

#### Client (if applicable)

- [x] UI changes look good on all screen sizes and browsers
- [x] No console errors or warnings
- [x] User experience and accessibility have been tested
- [ ] Added Storybook stories for new components
- [ ] Components follow design system guidelines (if applicable)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a new leaderboard overview component for improved display of leaderboard entries.
  - Added dynamic scrolling functionality to navigate to specific ranks within the leaderboard.
  - Enhanced the UI with a league info modal for additional league information.
  
- **Bug Fixes**
  - Improved rendering logic to ensure leaderboard data is displayed only when available.

- **Chores**
  - Removed the outdated leaderboard league component and its associated Storybook configuration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

milesha commented on webapp/src/app/home/leaderboard/leaderboard.component.ts:
```diff
@@ -53,6 +53,37 @@ export class LeaderboardComponent {
   leaderboard = input<LeaderboardEntry[]>();
   isLoading = input<boolean>();
 
+  protected adjustedLeaderboard = computed(() => {
```
We either need to come up with more precise naming than 'adjusted' or include clear code comments. Otherwise, the meaning of the functions will remain unclear without thoroughly reading through the codebase. 🤔

GODrums replied:
You are absolutely correct.
I just added a comment to clarify its purpose since I couldn't think of a fitting name.

milesha commented on webapp/src/app/home/leaderboard/leaderboard.component.ts:
```diff
@@ -53,6 +53,38 @@ export class LeaderboardComponent {
+  defaultSelfEntry = computed(() => {
+    return {
+      user: {
+        login: this.user()?.username,
```
`this.user()?` can be changed to `currentUser?` in the `defaultSelfEntry`, right? As we already have this variable saved

GODrums replied:
Not quite since `currentUser` is only a local variable in the computed-function of the leaderboard. I reworked the `defaultSelfEntry` in 0b90335 now with the user as parameter to realize your idea anyways.

milesha reviewed (dismissed):
lgtm, thanks for the improvements 👍🏼

FelixTJDietrich commented:
Looks for me like this, as it is now I think it kind of looks off
<img width="771" alt="image" src="https://github.com/user-attachments/assets/b9027821-e3b5-4b10-a2c9-f4b19dfb223b" />

FelixTJDietrich commented:
My proposal was to have a dedicated section displaying similar this information separate of the leaderboard but above, similarly to the profile header in the league PR. I don't like the duplication if the row.

FelixTJDietrich commented:
Actually just integrate it into the league section after it is merged 🤔

GODrums commented:
> My proposal was to have a dedicated section displaying similar this information separate of the leaderboard but above, similarly to the profile header in the league PR. I don't like the duplication if the row.

I tried doing that first, but it's super hard to get the alignments correctly when having a table row without any column headers. But I do see why duplicating the row looks bad for people at the top of the scoreboard. 
Maybe we instead want to try a different format in the league section?

FelixTJDietrich commented:
I mean thin one:
<img width="743" alt="image" src="https://github.com/user-attachments/assets/c22af64f-5308-491b-864c-3f0c843720f2" />

Why not integrate it there? It does not need to be the exact same as the table row, can be a slightly different layout. You still have your highlighted spot in the leaderboard once you scroll down 🤔

FelixTJDietrich reviewed (changes requested):
@GODrums how do we proceed with this PR? Would it be possible just to integrate it left of the current league process? 

There is anyways so much whitespace on the right. I don't want to change the order in the leaderboard. 
![image](https://github.com/user-attachments/assets/54fec51c-a190-453a-827d-c7c66949e456)

GODrums commented:
@FelixTJDietrich Understood! I tried to apply your suggestion and removed the original changes.

I added a small grid to display relevant leaderboard data about oneself: own score, current leaderboard rank and activity data.
Do let me know if this fits your idea better:
![image](https://github.com/user-attachments/assets/f02f675a-ba83-49fd-b353-c23972b3eda1)

FelixTJDietrich reviewed (approved):
I changed it to be like this now:
![image](https://github.com/user-attachments/assets/ab528034-c236-4279-a6f5-eed1f3d06b64)

Score and review activity are not so important there, I just think that the rank is important next to the avatar. You can click on it and then it scrolls you to the row.
